apps/stable_diffusion/scripts/txt2img.py

```diff
@@ -16,31 +16,62 @@ def main():
     if args.clear_all:
         clear_all()
 
+    # TODO: prompt_embeds and text_embeds form base_model.json requires fixing
     dtype = torch.float32 if args.precision == "fp32" else torch.half
     cpu_scheduling = not args.scheduler.startswith("Shark")
     set_init_device_flags()
     schedulers = get_schedulers(args.hf_model_id)
     scheduler_obj = schedulers[args.scheduler]
     seed = args.seed
-    txt2img_obj = Text2ImagePipeline.from_pretrained(
-        scheduler=scheduler_obj,
-        import_mlir=args.import_mlir,
-        model_id=args.hf_model_id,
-        ckpt_loc=args.ckpt_loc,
-        precision=args.precision,
-        max_length=args.max_length,
-        batch_size=args.batch_size,
-        height=args.height,
-        width=args.width,
-        use_base_vae=args.use_base_vae,
-        use_tuned=args.use_tuned,
-        custom_vae=args.custom_vae,
-        low_cpu_mem_usage=args.low_cpu_mem_usage,
-        debug=args.import_debug if args.import_mlir else False,
-        use_lora=args.use_lora,
-        use_quantize=args.use_quantize,
-        ondemand=args.ondemand,
-    )
+    if args.height == 1024:
+        assert (
+            args.width == 1024
+        ), "currently we support only 1024x1024 image size via SDXL"
+        assert args.precision == "fp16", "currently we support fp16 for SDXL"
+        # For SDXL we set max_length as 77.
+        args.max_length = 77
+        txt2img_obj = Text2ImageSDXLPipeline.from_pretrained(
+            scheduler=scheduler_obj,
+            import_mlir=args.import_mlir,
+            model_id=args.hf_model_id,
+            ckpt_loc=args.ckpt_loc,
+            precision=args.precision,
+            max_length=args.max_length,
+            batch_size=args.batch_size,
+            height=args.height,
+            width=args.width,
+            use_base_vae=args.use_base_vae,
+            use_tuned=args.use_tuned,
+            custom_vae=args.custom_vae,
+            low_cpu_mem_usage=args.low_cpu_mem_usage,
+            debug=args.import_debug if args.import_mlir else False,
+            use_lora=args.use_lora,
+            use_quantize=args.use_quantize,
+            ondemand=args.ondemand,
+        )
+    else:
+        assert (
+            args.height <= 768 and args.width <= 768
```

Comment:
I have an issue with splitting these up here based on the specified size.

The first thing people will do with an SDXL model is probably going to be: download SDXL model (or SDXL custom model) -> click on it in the UI dropdown -> enter prompt -> generate image

in this case they will get an error as it's set to 512x512 by default and so it just tries to use the `Text2ImagePipeline` instead of the `Text2ImageSDXLPipeline`.

Can we use the huggingface/civitAI model metadata to determine whether it is an SDXL-based model and then programmatically change the available sizes if yes? This technically works if you're aware of all the boxes to tick and dials to set, but if it breaks the first time you try it, that's bad for us.

I'm open to other points of view on this, but it could be as easy as getting the .ckpt filename and if it has XL we set some `is_sdxl` to True (False by default) and it takes the right path.


A considerable alternative to this is to move XL features to a separate `XL (Beta)` tab and then we won't have to worry about the funky stuff in the SD1.5/2.1 workflow that I ran into with this patch.